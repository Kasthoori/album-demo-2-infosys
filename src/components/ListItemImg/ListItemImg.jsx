import React from "react";
import Image from 'react-bootstrap/Image'

const ListItemImg = (props) => {

  // const handleClick = props.onclick;
   const thumbnail = props.track.thumbnailUrl;
   const title = props.track.title;
   const onItemClicked = props.onclick;

 
  return (
    
    <div data-testid="onclickevent"
      className="ListItemImg"
      onClick={() => {
        onItemClicked(props.track);
      }}
    >
      <Image data-testid="imagethumbnail" className="img-thumbnail" id="thumbnail" src={thumbnail} alt="" rounded />&nbsp;&nbsp;
      {/* <span id="text">{title}</span> */}
    </div>
    
  );
};

export default ListItemImg;