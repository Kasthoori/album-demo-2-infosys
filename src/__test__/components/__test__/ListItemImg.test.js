import React from 'react';
import { shallow } from 'enzyme';
import ListItemImg from '../../../components/ListItemImg/ListItemImg';
import toJson from 'enzyme-to-json';

      
     const mockFunction = jest.fn();

     const track = {
            thumbnailUrl : 'http://testurl',
            title : 'Test title',
       } 

describe('ListItemImg Component', () => {


    //test case 01. thumbnail image to tested
    test('test image thumbnail', () => {

        const wrapper = shallow(<ListItemImg onclick={mockFunction} track = {track} />);
         expect(wrapper.find('[data-testid="imagethumbnail"]').length).toBe(1);
      });

      //test case 02. number of tr tags
      test('number of tr tags', () => {

        const wrapper = shallow(<ListItemImg onclick={mockFunction} track = {track} />);
         expect(wrapper.find('div').length).toBe(1);
      });

      //test case 03. check onClick event call
        test('test button call', () => {
            const button = shallow(<ListItemImg onclick={mockFunction} track = {track}  />);
            button.find('[data-testid="onclickevent"]').simulate('click');
            expect(mockFunction.mock.calls.length).toEqual(1);
        });

    //render without crashing (snapshot)
    test('render without crashing', () => {
        const wrapper = shallow(<ListItemImg onclick={mockFunction} track = {track}  />);
        expect(toJson(wrapper)).toMatchSnapshot();
      
    
    });


});