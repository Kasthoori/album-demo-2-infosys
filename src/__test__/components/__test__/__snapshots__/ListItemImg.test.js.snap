// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemImg Component render without crashing 1`] = `
<div
  className="ListItemImg"
  data-testid="onclickevent"
  onClick={[Function]}
>
  <Image
    alt=""
    className="img-thumbnail"
    data-testid="imagethumbnail"
    fluid={false}
    id="thumbnail"
    rounded={true}
    roundedCircle={false}
    src="http://testurl"
    thumbnail={false}
  />
    
</div>
`;
